packages/docs/integrate/custom-integration.md

```diff
@@ -0,0 +1,334 @@
+
+```jsx
+// nuxt.config.js
+
+export default {
+  buildModules: ['~/integrations/jsonplaceholder/nuxt'],
+};
+```
+
+::: tip Why `buildModules` and not `modules`?
+
+In this example, you've created a very simple module responsible only for registering your integration's plugin. It does it during the application build, therefore its code does not need to be included in the final bundle.
+
+However, your integration's module could do way more that that. Visit the official Nuxt [documentation](https://nuxtjs.org/docs/directory-structure/modules/) to explore all capabilities of Nuxt modules and the difference between `modules` and `buildModules`.
+:::
+
+Secondly, add your integration's configuration in `middleware.config.js`:
+
+```jsx
+// middleware.config.js
+
+const path = require('path');
+
+module.exports = {
+  integrations: {
+    jsonplaceholder: {
+      location: path.resolve(
+        __dirname,
+        'integrations/jsonplaceholder/src/index.server.js'
+      ),
+      configuration: {
+        baseURL: 'https://jsonplaceholder.typicode.com/',
+        // apiKey: '1234'
+      },
+    },
+  },
+};
+```
+
+In this example, the configuration will only contain `baseURL`. However, if your integration requires passing some sensitive credentials (like tokens or API keys), this is the right place to put them.
+
+You might be wondering about the `location` part of the above configuration, and you are right - it refrences a file that does not exist yet. Do not worry. You're going to create it in the next step.
+
+## Creating an API client
+
+Your integration will need an API client to communicate with the 3rd party. For most services, you would use a client provided by their SDKs. This is what Vue Storefront does for the majority of our integrations (e.g. [Contentstack](https://www.contentstack.com/docs/developers/javascript-browser/api-reference/)).
+
+For [JSON Placeholder Free Fake API](https://jsonplaceholder.typicode.com/), we're simply going to create a dedicated [axios](https://github.com/axios/axios) instance. Let’s start with adding the missing `index.server.js` file in `/integrations/jsonplaceholder/src`.
+
+```jsx
+// integrations/jsonplaceholder/src/index.server.js
+
+const { apiClientFactory } = require('@vue-storefront/core');
+const api = require('./api');
+const axios = require('axios');
+
+const setup = (settings) => {
+  const { baseURL } = settings;
+
+  const client = axios.create({ baseURL });
+
+  return {
+    config: {
+      ...settings,
+    },
+    client,
+  };
+};
+
+const { createApiClient } = apiClientFactory({
+  onCreate: setup,
+  api,
+});
+
+module.exports = { createApiClient };
+```
+
+The `setup` function is crucial here because this is where we create our `axios` client. It receives `baseURL` we had specified in `middleware.config.js` and will be used by API methods located in `/integrations/jsonplaceholder/src/api.js`. Let's create the file as the next step of this tutorial.
```

Comment:
```suggestion
The `setup` function is crucial because it's where we create our `axios` client. It receives the `baseURL` we specified in `middleware.config.js` and will be used by API methods in `/integrations/jsonplaceholder/src/api.js`. Let's create the file as the next step of this tutorial.
```